accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,277 @@
+- Feature Name: CaaSP nodes action locking
+- Start Date: 2019-10-28
+- RFC PR: TBD-change filename
+
+# Summary
+[summary]: #summary
+
+This RFC describes the special handling that Uyuni/SUSE Manager must implement to manage CaaS Platform nodes.
+
+# Motivation
+[motivation]: #motivation
+
+<!-- - Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions. -->
+
+Any node of a CaaS Platform cluster can be registered to Uyuni/SUSE Manager.
+Registering a CaaS Platform node to Uyuni/SUSE Manager brings the following notable advantages:
+
+- The user can see the patch level status of the CaaS Platform nodes (at the packages level, not at the container level - the latter is called "[container staging](https://confluence.suse.com/display/~dmacvicar/Container+Staging+Breakout)" and it is outside of the scope of this RFC)
+- The user can perform configuration management operations using Salt from Uyuni/SUSE Manager
+- The user can assign a different set of channels (with channel staging and/or CLM filters) to different clusters and manually control from Uyuni/SUSE Manager which version of CaaS Platform gets automatically installed in its CaaSP clusters<sup>1</sup>.
+
+However, if the user does some specific actions on any node of the cluster via Uyuni/SUSE Manager, the cluster may result temporarily or permanently unusable or broken.
+This RFC describes how Uyuni/SUSE Manager can prevent the user to perform these kinds of actions on the CaaS Platform nodes.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+## Terminology
+
+In this RFC, we will use:
+
+* CaaS Platform node: any node that comprises the cluster, either a control plane node or a worker node
+* Bootstrap/registration in the Uyuni/SUSE Manager context: registering a node to Uyuni/SUSE Manager via [the documented methods](https://opensource.suse.com/doc-susemanager/suse-manager/client-configuration/registration-overview.html).
+At the time of writing, registering a CaaS Platform node to Uyuni/SUSE Manager has to be manually performed by the user with:
+  * Bootstrap script
+  * UI: Systems > Bootstrapping
+  * `bootstrap` XMLRPC call
+  * Not usual but still possible and hackier: mass bootstrap via salt-ssh and a user-generated roster file
```

Comment:
Will plain key acceptance never work? Why?

Reply:
That could work as well, but it not published in the documentation as an "official" bootstrap method.
Do you think it would make sense to add it here?

Reply:
Yes.

But IIUC that would mean the list would comprise all possible registration methods we have, so maybe you could even consider getting rid of the list altogether and just say "with one of the available registration methods".